packages/extension-ui/src/Popup/Signing/Extrinsic.tsx

```diff
@@ -26,6 +26,32 @@ interface Props {
   url: string;
 }
 
+interface AssetId { parents: number, interior: { 'X2': [{ PalletInstance: string }, { GeneralIndex: string }] } }
+
+function isAssetId (assetId: unknown): assetId is AssetId {
+  if (!assetId || typeof assetId !== 'object') {
+    return false;
+  }
+
+  return 'parents' in assetId && 'interior' in assetId &&
+    typeof assetId.interior === 'object' && assetId.interior !== null && 'X2' in assetId.interior &&
+    Array.isArray(assetId.interior.X2) && assetId.interior.X2.length === 2 &&
+    'PalletInstance' in assetId.interior.X2[0] &&
+    'GeneralIndex' in assetId.interior.X2[1];
+}
+
+function decodeAssetId (assetId: unknown) {
+  if (!isAssetId(assetId)) {
+    return null;
+  }
+
+  try {
```

Comment:
For the return value of the AssetId I think we can just return the whole `MultiLocation`, and shouldn't extrapolate any specific, this will keep it true to the original input. 

cc: @IkerAlus Thoughts?

Reply:
Why are we limiting this to only asset IDs as XCMV2 junction objects with the junctions `PalletInstance` and `GeneralIndex`? Is there a limitation I am not aware of?

From UX perspective, indeed, most of the cases of users paying fees in other tokens beyond will have an asset ID in the mentioned form, but actually any asset with an existing pool against DOT (asset conversion pallet) will be able to do this, for example any foreign asset sent to AH. 

The general solution will be to allow any `Multilocation` as asset ID.

Reply:
We thought of this but I think this XCMV2 object covers 99% of usecases. This change doesn't affect the actual functionality, so if the user provides say and interior with length 3/4/5/6/7/8, it will still go through. This function is just to display it in the UI. 

For the other types, its ambigious how it would be displayed in the UI. should we just assume the last element in the array is always `GeneralIndex` and it is the asset id?

Reply:
I see, thanks! Indeed, it makes sense to only show these cases in the UI, particularly considering it won't crash and simply not show the general case.

Reply:
Isn't the whole multilocation considered the Asset though? So should we show the whole multilocation?